src/OrbitGl/TimeGraph.cpp

```diff
@@ -159,24 +159,15 @@ void TimeGraph::VerticalZoom(float zoom_value, float mouse_normalized_y_position
   const float old_scale = layout_.GetScale();
   layout_.SetScale(old_scale / ratio);
 
-  // We are limiting the maximum and minimum zoom-level, so the real ratio could be different.
-  const float capped_ratio = old_scale / layout_.GetScale();
-
-  const float world_height = viewport_->GetVisibleWorldHeight();
-  const float mouse_position_in_world_y =
-      viewport_->GetScreenTopLeftInWorld()[1] + mouse_normalized_y_position * world_height;
-  const float relative_mouse_position_in_world_y =
-      mouse_position_in_world_y - viewport_->GetScreenTopLeftInWorld()[1];
-
-  // We define world system coordinates in relation of the screen. As we are zooming on the screen,
-  // the world is being expanded or collapsed. Therefore the mouse position in world should be
-  // adapted in the same way.
-  const float new_mouse_position_in_world_y = mouse_position_in_world_y / capped_ratio;
-
-  float new_screen_top_left_in_world_y =
-      new_mouse_position_in_world_y - relative_mouse_position_in_world_y;
-
-  viewport_->SetScreenTopLeftInWorldY(new_screen_top_left_in_world_y);
+  // Adjust the scrolling offset such that the point under the mouse stays the same if possible.
+  // For this, calculate the "global" position (including scaling and scrolling offset) of the point
+  // underneath the mouse with the old and new scaling, and adjust the scrolling to have them match.
+  const float offset_from_top_in_world = viewport_->GetWorldHeight() * mouse_normalized_y_position;
+  const float total_mouse_y_position =
```

Comment:
nit. I find "total_mouse_y_position" confusing. What does it mean for a position to be "total"? Maybe simply using "mouse_y_position" could be enough?